### DISCUSSION TITLE
AWS CodeCommit: Not able to clone repository

### DISCUSSION THREAD

**Discussion by blattr** [codemagic-ci-cd/codemagic-docs#1424] [Q&A] [answered]
Hello everyone,

the last successful build of my (business) application was 4 days ago (id=63ef6bf5b44082171441493c). Since then, nothing changed in buildspec.yaml, any integration or credentials (on any side). For an unknown reason, I am not able to clone my repository on AWS CodeCommit anymore. The last failed build was just 30 minutes ago (id=63f4b9efc698fb64afb8ec19). I am connecting to AWS CodeCommit using HTTPs (username/password).

This is the error message (removed personal information):

_== Cloning from feature branch 'master' ==
\> /usr/bin/git clone -b master --depth 50 https://codemagic-at-123@git-codecommit.eu-central-1.amazonaws.com/v1/repos/abc /Users/builder/clone --progress
Cloning into '/Users/builder/clone'...
fatal: unable to access 'https://git-codecommit.eu-central-1.amazonaws.com/v1/repos/abc/': The requested URL returned error: 403_

I also have a personal account with exact the same setup - it stopped working as well. If needed, I can provide the build ids as well of course.

**Comment by blattr** [comment]
> Hello everyone,
> 
> the last successful build of my (business) application was 4 days ago (id=63ef6bf5b44082171441493c). Since then, nothing changed in buildspec.yaml, any integration or credentials (on any side). For an unknown reason, I am not able to clone my repository on AWS CodeCommit anymore. The last failed build was just 30 minutes ago (id=63f4b9efc698fb64afb8ec19). I am connecting to AWS CodeCommit using HTTPs (username/password).
> 
> This is the error message (removed personal information):
> 
> _== Cloning from feature branch 'master' == > /usr/bin/git clone -b master --depth 50 https://codemagic-at-123@git-codecommit.eu-central-1.amazonaws.com/v1/repos/abc /Users/builder/clone --progress Cloning into '/Users/builder/clone'... fatal: unable to access 'https://git-codecommit.eu-central-1.amazonaws.com/v1/repos/abc/': The requested URL returned error: 403_
> 
> I also have a personal account with exact the same setup - it stopped working as well. If needed, I can provide the build ids as well of course.

Update: I switched to SSH. Now it is working again. Even after a deep dive into the logs I did not find the source of the problem or a workaround.

**Reply by dzero1** [reply]
How did you switched to SSH ? I tried to change the repo url to use ssh, but it's always show an error.

**Reply by blattr** [reply]
@icarusdust wrote about the (more or less) exact same thing I did to solve this issue. ssh-keygen is the command line tool I used to generate the key pair (macOS).

To add a little bit of information - SSH-Key-ID is the ID you get, when you upload the private key to your AWS IAM user.

**Comment by dzero1** [comment]
I'm confirming this. I have similar problem. Mine was bult correctly 5days before. But not now.

**Comment by icarusdust** [comment, marked as answer]
@dzero1, SSH url should look like this:
`ssh://xxxxxxxxxx@git-codecommit.us-west-2.amazonaws.com/v1/repos/blabla_app.git`
And that XXXXX is your SSH-Key-ID. In the meantime, i could reproduce the issue locally with cloning a repo from codecommit via https.
